test_mesh.py

```diff
@@ -0,0 +1,17 @@
+import numpy as np
+import pytest
+
+from generate_mesh import generate_mesh
+
+def test_generic():
+	expected = np.array([0, 0.5, 1])
+	layer_outer_radii = [0, 1]
+	nodes_per_layer = [3]
+	
+	assert np.all(expected == generate_mesh(layer_outer_radii,nodes_per_layer))
+	
+def test_several_materials():
+   layer_outer_radii = [0, 1, 2, 3]
+   nodes_per_layer= [3, 2, 5]
+   expected = np.array([0, 0.5, 1, 1, 2, 2, 2.25, 2.5, 2.75, 3])
```

Comment:
Are you sure that you want repeated mesh values?

Reply:
The way I planned on addressing the boundaries between layers requires having a mesh point for each material.

Reply:
Are you planning to have an interface thermal resistance?

Reply:
Is dT/dx or q still continuous at the boundary in the case of interface thermal resistance? How will the second derivative be formulated for the "points" that both represent the boundary?

Reply:
The continuity condition would be that Temperature is constant and heat flux ( -k*dt/dx) is constant across the boundary. It may be that having a single node at the boundary is sufficient, this is just the way I approached it initially.

Reply:
I think it will be complicated to generate your matrix with these repeated points and that matrix may be singular (unable to be solved), but I'm not certain. It may depend on how you try to do it, things may work out.